src/server/pkg/storage/chunk/storage.go

```diff
@@ -17,20 +17,24 @@ const (
 
 // Storage is the abstraction that manages chunk storage.
 type Storage struct {
+	memObj    obj.Client
```

Comment:
Why does chunk storage need two object storage clients? Is it ever going to operate on the one without the memory caching wrapper?

Reply:
They don't form a single storage hierarchy.  The memory store holds uncompressed cipher text.  The disk cache and object storage form a storage hierarchy for ciphertexts.